appservice/src/deploy/localGitDeploy.ts

```diff
@@ -74,7 +74,7 @@ export async function localGitDeploy(client: SiteClient, fsPath: string, context
                         // for whatever reason, is '-f' exists, true or false, it still force pushes
                         const pushOptions: git.Options = forcePush ? { '-f': true } : {};
 
-                        localGit.push(remote, 'HEAD:master', pushOptions).catch(async (error) => {
+                        localGit.push(remote, `HEAD:${status.current}`, pushOptions).catch(async (error) => {
```

Comment:
I think that this will break the current experience.  The remotes that exist in Azure itself only have a master branch.  That is, Azure only looks at the master branch when kudu builds the website on the remote.

If I recall correctly, HEAD:master means use your current branch, and push it the remote master branch.

Reply:
Darn I thought I had tested and it worked still for normal apps. I just tested and yeah it breaks the current experience. I think I'll add a parameter to `localGitDeploy` in which you can specify the branch.

Reply:
Is this behavior different in trial apps?  I'd figure that you would just deploy to the trial app's master branch as well.

Reply:
Ya it has to be HEAD:RELEASE to work for trial apps.